server/data/data-library/src/main/java/org/apache/james/rrt/lib/AbstractRecipientRewriteTable.java

```diff
@@ -297,6 +299,11 @@ public void removeGroupMapping(MappingSource source, String address) throws Reci
         removeMapping(source, mapping);
     }
 
+    @Override
+    public List<MappingSource> listSources(Mapping mapping) throws RecipientRewriteTableException {
+        return null;
```

Comment:
I'm pretty sure it is not what is expected... Maybe we should filter the all Mappings content here...

Reply:
IMO, why we should filter all Mappings content instead of using query in each of implementation?

Reply:
> why we should filter all Mappings content instead of using query in each of implementation?

I do not fully understand your question, but the goal would be to provide a default, correct, but not efficient behaviour by default.

This enables:

 - Implementers of RecipientRewriteTable to not have their implementation broken because of a missing method in the implementation
 - Allows working on the webadmin API without having to wait for an implementation to implement this API.